### PR TITLE
DDPB-4735 split out frontend container

### PR DESCRIPTION
**Pull request by jamesrwarren**
## Purpose
Split out frontend container and end TLS at the LB

Fixes DDPB-4735

## Approach
This is a fairly hefty refactor with a fairly hefty amount of changes. Might be worth writing out the logic here so it can be understood before reviewing. Happy to change some bits around or names of things!

Previous folder structure:

image

We move this under client/app and split out the docker and confd setup into a docker folder and the webpack/js/css frontend stuff into a folder called resources. Structure looks like this:

image

We add a container for building the node resources that maps to a folder called public locally and use the docker volume mapping to map the public folder into both the nginx container and the app container.

For CI, we build the frontend web resources as a separate runner and bring them in as an artifact for client (app) and client-webserver (nginx). I normally like to have all my bits like tests and builds running in containers so errors are reproducible locally (even if this means slightly slower builds) but in this case it was too much of a hit. I manage drift by extracting the node version to use from the dockerfile so it only needs updating in one place.

We then modify the task definition so that it includes both containers and amend firewall sg rules to allow port 80 from LB to the app. 

ISSUE: for some unknown reason the integration tests are sometimes going via http rather than https when pressing buttons. As such I have had to allow port 80 to LB and added rerouting to 443 from 80 for both LBs. I have tried printing out current urls and base urls in all the places it sets this as part of the tests and it always seems to go to https according to that and yet somehow it is not! :(

We modify the nginx config to manage the new setup and make sure we have the same security headers as well as some additional ones that I believed were missing. Took the opportunity to harden the containers for additional security.

That is the crux of the PR. 

ISSUE: The integration tests seem slower in some places and faster in others. Would be nice to work out why.

## Learning
NA

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [ ] I have added tests to prove my work
- [ ] The product team have approved these changes
- [ ] I have checked my work for potential security issues and refered to the [OWASP top 10](https://owasp.org/www-project-top-ten/)

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete
- [ ] If adding GA events, I have updated or [checked](docs/runbooks/GOOGLE-ANALYTICS.md) the existing category or label values

### CODE REVIEW

File: client/app/config/services.yml [hunk 1]
```diff
@@ -260,7 +260,7 @@ services:
         base_uri: "%api_base_url%"
         verify: false
         headers:
-          Content-Type: application/json
+          Content-Type: app/json
```
Comment (mattmachell) [inline client/app/config/services.yml:263]: Find and replace issue?

Comment (jamesrwarren) [reply client/app/config/services.yml:263]: Yeah, this isn't ready for review yet. That button shouldn't have been clicked.

Comment (jackgoodby) [reply client/app/config/services.yml:263]: Mea culpa - Jim and I misunderstood each other

File: .github/workflows/_web-resources.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        uses: actions/cache@04f198bf0b2a39f7230a4304bf07747a0bddf146 # pin@v3.0.8
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ./web-resources/public
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/web-resources/package-lock.json') }}
+      - name: build assets
+        run: |
+          # Copy in templates and translations
+          cp -r ../app/templates ./templates
```
Comment (mattmachell) [inline .github/workflows/_web-resources.yml:29]: Noticed this in passing. Templates are PHP/Twig code so shouldn't be needed in the node assets or the web server.

Comment (jamesrwarren) [reply .github/workflows/_web-resources.yml:29]: yeah this was in the build image from before. I thought it seemed strange and tried removing it but it was still expecting a file somewhere... I must have been misunderstanding the message though as I have now removed that cp in and it still seems fine as you would expect :)

**Review by Gugandeep — APPROVED** [review]

**Pull request merged**